apis/Google.Cloud.Logging.V2/Google.Cloud.Logging.Log4Net/LruCache.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Logging.Log4Net
+{
+    class LruCache<K, V>
+    {
+        struct Item
+        {
+            public Item(LinkedListNode<K> node, V value)
+            {
+                Node = node;
+                Value = value;
+            }
+
+            public LinkedListNode<K> Node { get; }
+            public V Value { get; }
+        }
+
+        public LruCache(int capacity)
+        {
+            _capacity = capacity;
+            _dict = new Dictionary<K, Item>(capacity);
+            _order = new LinkedList<K>();
+        }
+
+        private readonly object _lock = new object();
+        private readonly int _capacity;
+        private readonly Dictionary<K, Item> _dict;
+        private readonly LinkedList<K> _order;
+
+        public V GetOrAdd(K key, Func<V> valueFn)
```

Comment:
I see why you use func now.  As far as this is not a generic purpose class, I think it's OKay to keep it this way --- assuming very low change others use this class in other way.